spark/v3.1/spark/src/main/java/com/netease/arctic/spark/table/ArcticSparkChangeTable.java

```diff
@@ -18,28 +18,151 @@
 
 package com.netease.arctic.spark.table;
 
-import org.apache.iceberg.Table;
+import com.netease.arctic.io.ArcticFileIO;
+import com.netease.arctic.op.OverwriteBaseFiles;
+import com.netease.arctic.op.RewritePartitions;
+import com.netease.arctic.scan.KeyedTableScan;
+import com.netease.arctic.table.BaseKeyedTable;
+import com.netease.arctic.table.BaseTable;
+import com.netease.arctic.table.ChangeTable;
+import com.netease.arctic.table.KeyedTable;
+import com.netease.arctic.table.MetadataColumns;
+import com.netease.arctic.table.PrimaryKeySpec;
+import com.netease.arctic.table.TableIdentifier;
+import org.apache.iceberg.HasTableOperations;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
-import org.apache.iceberg.spark.source.SparkTable;
+import org.apache.iceberg.types.Types;
 import org.apache.spark.sql.connector.catalog.TableCapability;
-import org.apache.spark.sql.types.StructType;
 
+import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ArcticSparkChangeTable implements KeyedTable, HasTableOperations {
```

Comment:
if ArcticSparkChangeTable is not subclass of SparkTable.  time travel feature is disable.